packages/lesswrong/components/posts/PostsPage/PostsPage.tsx

```diff
@@ -774,7 +774,7 @@ const PostsPage = ({fullPost, postPreload, eagerPostComments, refetch, classes}:
             />}
             {isAF && <AFUnreviewedCommentCount post={post}/>}
           </AnalyticsContext>
-          {isFriendlyUI && post.commentCount < 1 &&
+          {isFriendlyUI && Math.max(post.commentCount, results?.length ?? 0) < 1 &&
```

Comment:
This fixes a bug where the "This post has no comments" message remains on screen after a user adds the first comment